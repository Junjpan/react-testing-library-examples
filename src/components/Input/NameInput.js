import React, { useState } from "react";
import "./nameinput.css";

const NameInput = () => {
  const [name, setName] = useState("");

  return (
    <div>
      <span data-testid='change-input-greeting'>
        Welcome, {name}
      </span>
      <br/>
      <input
        type='text'
        className='input'
        placeholder='Please enter your name'
        aria-label='nameinput'
        value={name}
        onChange={(e) => setName(e.target.value)}
      ></input>
    </div>
  );
};

export default NameInput;
