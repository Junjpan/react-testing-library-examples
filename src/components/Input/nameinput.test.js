import React from 'react';
import ReactDOM from 'react-dom';
import NameInput from './NameInput';
import {render,fireEvent,cleanup} from '@testing-library/react';

test('Should render without a crash',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<NameInput />,div)
})

test('Should display correct greeting',()=>{
    const {getByLabelText,getByTestId}=render(<NameInput />)
    const input=getByLabelText('nameinput');
    const greeting=getByTestId('change-input-greeting');
    expect(input.placeholder).toBe('Please enter your name');
    expect(input.value).toBe('');
    expect(greeting.textContent).toBe('Welcome, ')
    fireEvent.change(input,{target:{value:"testing"}});
    expect(input.value).toBe('testing');
    expect(greeting.textContent).toBe('Welcome, testing');

})

test('it match the snapshot',()=>{
    const {asFragment}=render(<NameInput />);
    expect(asFragment()).toMatchSnapshot()
})