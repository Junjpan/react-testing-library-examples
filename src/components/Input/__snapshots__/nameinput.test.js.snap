// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <span
      data-testid="change-input-greeting"
    >
      Welcome, 
    </span>
    <br />
    <input
      aria-label="nameinput"
      class="input"
      placeholder="Please enter your name"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
